model/src/main/kotlin/AdvisorRecord.kt

```diff
@@ -49,8 +56,21 @@ data class AdvisorRecord(
      * True if any of the [advisorResults] contain [OrtIssue]s.
      */
     val hasIssues by lazy {
-        advisorResults.any { advisorResultContainer ->
-            advisorResultContainer.results.any { it.summary.issues.isNotEmpty() }
+        advisorResults.any { (_, results) ->
+            results.any { it.summary.issues.isNotEmpty() }
+        }
+    }
+}
+
+private class AdvisorResultsDeserializer : StdDeserializer<SortedMap<Identifier, List<AdvisorResult>>>(
```

Comment:
Just a general idea for later: As we have several such custom deserializers by now to maintain backward compatibility, how about moving all of them to a central `BackwardCompatibility.kt` to get an overview over how much technical debt we carry around?

Reply:
That would also allow us to move legacy classes, like now `AdvisorResultContainer`, to that file and make them private to avoid accidental use in current code.

Reply:
And can't we remove `AdvisorResultContainerTest.kt` altogether then?

Reply:
> As we have several such custom deserializers by now to maintain backward compatibility, how about moving all of them to a central BackwardCompatibility.kt to get an overview over how much technical debt we carry around?

Maybe, but I think there is also some backward compatibility code that is not separated in a class, and we would probably need such a file in other modules as well. Using `@Deprecated` or a custom annotation would be another approach.

Reply:
> And can't we remove `AdvisorResultContainerTest.kt` altogether then?

I think it should be removed together with the class.